code/modules/vehicles/pimpin_ride.dm

```diff
@@ -47,7 +50,7 @@
 		floorbuffer = TRUE
 		qdel(I)
 		to_chat(user, "<span class='notice'>You upgrade [src] with the floor buffer.</span>")
-		flags_1 |= CLEAN_ON_MOVE_1
+		LoadComponent(/datum/component/cleaning)
```

Comment:
AddComponent

Reply:
Also a few lines up should check for the cleaning component instead of floorbuffer

Reply:
But we want to keep the floorbuffer var for drawing the sprite?

Reply:
Just the check, not the var itself